### PR TITLE
AWS system test example_dynamodb_to_s3: add retry when fecthing the export time

### PR DESCRIPTION
**Pull request by vincbeck**
AWS Dynamodb [UpdateContinuousBackups](https://docs.aws.amazon.com/amazondynamodb/latest/APIReference/API_UpdateContinuousBackups.html) API is not stable and we need to add a retry logic when calling `update_continuous_backups`. The API might return `Backups are being enabled for the table: <...>. Please retry later`.

As long as I dont like doing that, we have no choice since the API behavior is not stable on the service side ...

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: airflow/providers/amazon/aws/utils/__init__.py
Comment (vandonr-amz) [inline airflow/providers/amazon/aws/utils/__init__.py]: why the random value ? a fixed number of seconds is not good ?

Comment (vincbeck) [reply airflow/providers/amazon/aws/utils/__init__.py]: That's part of backoff algorithm specification :)

Comment (vandonr-amz) [reply airflow/providers/amazon/aws/utils/__init__.py]: which spec ? What's the purpose ? It seems a random value is necessary when several agents are using the same strategy to access a resource and can collide, but it's not the case here ?

Comment (vincbeck) [reply airflow/providers/amazon/aws/utils/__init__.py]: You can find multiple documentation/specifications on Google:
- [Example](https://cloud.google.com/iot/docs/how-tos/exponential-backoff#:~:text=An%20exponential%20backoff%20algorithm%20retries,seconds%20and%20retry%20the%20request.)
- [Example](https://en.wikipedia.org/wiki/Exponential_backoff)

I am just trying to use an existing algorithm and have an implementation for it. I dont think we should modify the algorithm here. But if this is something you are really against I can remove it, I just dont see the harm here

Comment (o-nikolas) [reply airflow/providers/amazon/aws/utils/__init__.py]: I don't have any concern with this random delay. But there is also a [python package](https://pypi.org/project/retry/) for retrying so that we don't have to worry about implementing and maintaining any retry logic at all. That could be a good option.

**Review by o-nikolas — APPROVED** [review]

File: airflow/providers/amazon/aws/utils/__init__.py
Comment (vandonr-amz) [reply airflow/providers/amazon/aws/utils/__init__.py]: Yeah I read the wikipedia page, and the random part is only mentioned in "collision avoidance", which does not seem relevant here. It's not problematic per say as it will work properly like this, but it's just unnecessary.
Using the python package would be nice too, but it means adding a new dependency, which may be more trouble ?

**Review by vandonr-amz — APPROVED** [review]
ok to let it go like this, but I'd prefer it without the random.

**Review by vandonr-amz — APPROVED** [review]

**Comment by vincbeck** [comment]
Thanks for the feedbacks guys, it made me look if any library was already used for such purpose and there is one: `tenacity`. I removed the custom implementation and now use this library. It should address all your concerns.

**Review by o-nikolas — APPROVED** [review]
Approved, but I'll hold off merging since there is a broken static check making the build go red.

**Comment by ferruzzi** [comment]
@o-nikolas  and @vincbeck  the CI is green on this one, do you need another review on it or is it good to go?

**Comment by o-nikolas** [comment]
> @o-nikolas and @vincbeck the CI is green on this one, do you need another review on it or is it good to go?

Nope, I've just been merging/retrying/waiting for it to be green today. I'll merge now :)

**Pull request merged**
